### DISCUSSION TITLE
[2D Map] Hunting area map layer ?

### DISCUSSION THREAD

**Discussion by mff73** [mytourbook/mytourbook#963] [Q&A]
Hello,
It is hunting season in France, and I like hiking without being afraid 😨.
I use this map online which shows none hunting areas (reserve de chasse)
http://www.observatoire.savoie.equipement-agriculture.gouv.fr/Communes/rcfs.php
Is there a way for any expert, to add this layer in MTB above any 2D map? I looked but I can't find anything myself.
Thanks.

**Comment by wolfgang-ch** [comment]
When a map or overlay is from a tile server, like https://mytourbook.sourceforge.io/mytourbook/index.php/map-provider then it can be overlayed over the 2D map

**Comment by FJBDev** [comment]
After looking at the webpage, I see that they offer [here](http://catalogue.geo-ide.developpement-durable.gouv.fr/catalogue/srv/fre/catalog.search;jsessionid=1C43D2B08EAB95A672BFAC780CCB48AF.tc_geoide-catalogue-fo-inter_171_25#/metadata/fr-120066022-wmsmap-srv-27bfc02e-51db-4011-a7de-48bf1dd34459) a WMS link.

`https://carto2.geo-ide.din.developpement-durable.gouv.fr/rest-api/ows/10c4dcf8-f0a8-417e-91d1-c3b88e15b2fc?request=GetCapabilities&version=1.3.0&service=WMS`
@wolfgang-ch , can a WMS provider be overlayed in MT ?

**Reply by wolfgang-ch** [reply]
> @wolfgang-ch , can a WMS provider be overlayed in MT ?

Yes ...

![image](https://user-images.githubusercontent.com/1283445/201604069-72026e88-0383-4823-b8dc-d70d969c379b.png)

...but there are issues

* Latitude/longitude seems to be exchanged
* Tiles are looking a bit scrambled

![image](https://user-images.githubusercontent.com/1283445/201604300-a0843a0a-5521-4c0a-831e-599871c8389b.png)

**Reply by FJBDev** [reply]
> ...but there are issues
> 
>     * Latitude/longitude seems to be exchanged
> 
>     * Tiles are looking a bit scrambled
> 
> 
I noticed the same issue when I worked on [this bug](https://github.com/mytourbook/mytourbook/issues/949) but because it was the first time I was working on WMS import and because the code seems old and therefore tested and used a lot, I thought it was my layer that was bad.

**Reply by wolfgang-ch** [reply]
The WMS map provider is the less maintained map provider, I was wondering that it was still working, more or less with an age of 10+ years 😄 

When I created it, I was aware that there are many features which are not supported but maybe it can be refurbished by fixing the above bugs.

**Reply by FJBDev** [reply]
I noticed that it was an old feature as [the documentation](https://mytourbook.sourceforge.io/mytourbook/index.php/map-provider/wms) also has the below limitations:

```
Limitations

    the projection which is currently supported is EPSG:4326
    only the image size of 256 pixel is supported when the wms map provider is used within a profile map provider
```

>When I created it, I was aware that there are many features which are not supported but maybe it can be refurbished by fixing the above bugs.

I have never had the need to use this feature yet but it looks like a useful one

**Reply by mff73** [reply]
Hey, thanks for all your investigations. Not sure to understand everything.
If too complicated, please, don't bother, I will surely survive to hunters 🤣.
Btw, thanks Fred, I put "as is" the wms url you found, in Locus Map. Something is displayed 😁, but not same areas as on the website 🤔.

**Reply by FJBDev** [reply]
> Hey, thanks for all your investigations. Not sure to understand everything. 

No problem. Essentially, Wolfgang and I are looking at the WMS feature that could potentially display your hunter map but the WMS feature might need some improvements first.
But even with the WMS feature improved, I don't see the "cartes de chasse" an option in the list of layers...
The closest I see is `N_RES_NATURELLE`

![image](https://user-images.githubusercontent.com/25235665/201775415-8321646d-4e1b-446b-a5c4-9152cfd27ab4.png)


> If too complicated, please, don't bother, I will surely survive to hunters 🤣. Btw, thanks Fred, I put "as is" the wms url you found, in Locus Map. Something is displayed 😁, but not same areas as on the website 🤔.

What do you mean by `Locus Map` ?

**Comment by wolfgang-ch** [comment]
The Hunting area is the RCFS layer, when I delete it, the hunting areas disappear

![image](https://user-images.githubusercontent.com/1283445/201836740-10ed49a7-6933-49f2-8e9c-809b71681db3.png)


When observing all loaded resources in the browser, then the map images are from a wms tile server but the hunting areas are not from tile images.

![image](https://user-images.githubusercontent.com/1283445/201837024-fba48424-e3ab-4c19-88dc-45d7e365473b.png)

**Reply by FJBDev** [reply]
> The Hunting area is the RCFS layer, when I delete it, the hunting areas disappear

What i meant is that i dont see this RCFS layer in the list of WMS layers list.

**Comment by mff73** [comment]
@Fred,
When I speak about Locus Map, it is Locus Map android application, which I use a lot, and it has a WMS functionality.
Thanks to your url discovered, I can see layers, but like you, not RCFS 😕
Here are Locusmap screen shot.
![SmartSelect_20221115-211031_Locus Map](https://user-images.githubusercontent.com/50132146/202016422-eecfc6e8-7227-4365-ab8b-61351531a2ba.jpg)
![SmartSelect_20221115-211053_Locus Map](https://user-images.githubusercontent.com/50132146/202016450-bf1922e3-625e-42ae-b843-3dcc3d095cf6.jpg)

**Reply by FJBDev** [reply]
> @fred, When I speak about Locus Map, it is Locus Map android application, which I use a lot, and it has a WMS functionality. Thanks to your url discovered, I can see layers, but like you, not RCFS 😕 Here are Locusmap screen shot. 

Ah ok, I didn't know Locus Map.

Thanks for confirming, I don't see the RCFS anywhere so we need to find out where it comes from (a tile? a WMS? which WMS?)

**Comment by FJBDev** [comment]
When I look at the console, it seems that there are several zip files downloaded and 1 specifically for RCFS

![image](https://user-images.githubusercontent.com/25235665/202035705-c44ff7dd-9502-4686-a86f-0e4caa432e52.png)

You can download it [here](http://www.observatoire.savoie.equipement-agriculture.gouv.fr/Communes/rcfs.zip) but it doesn't seem to be a tile information file that MT can read

**Comment by wolfgang-ch** [comment]
The zip file contains rcfs.dbf with the historic dBase file format which can be opened in LibreOffice. 

There is also rcfs.shp which could be a shape file https://gisgeography.com/arcgis-shapefile-files-types-extensions/ with the hunting areas.

Both files are currently not supported in MT to display them in the 2D map. There may be somewhere a tile server which could convert shape files into pixel images and they could be displayed/overlayed in MT.

![image](https://user-images.githubusercontent.com/1283445/202093709-f1a52f18-e7c1-4601-84f4-2ac368d4faa1.png)

**Reply by FJBDev** [reply]
It's the first time I discover this `dbf` format. I learn new stuff every day 😄

**Reply by wolfgang-ch** [reply]
This shows that you are a younger generation

dBase and [Clipper](https://en.wikipedia.org/wiki/Clipper_(programming_language))  was my first database which I programmed in the 1980s within [MS-DOS](https://en.wikipedia.org/wiki/MS-DOS), many years before Windows or OS/2 appeared

**Reply by FJBDev** [reply]
> the 1980s within [MS-DOS](https://en.wikipedia.org/wiki/MS-DOS), many years before Windows or OS/2 appeared

Correct, I was not born in 1980 ! 😃

**Comment by mff73** [comment]
thanks Sherlock and Watson (who is who 😁).
I downloaded the rcfs.zip, i uploaded all files here https://mygeodata.cloud/converter/shp-to-gpx and i obtain rcfs.gpx which are the hunting zones (hunting not authorized) converted into tours.
not perfect, but it gives me the opportunity to see them in MTB above some of my tours 👍👍
Effectively, something like tile server could be really better, but if it doesn't work like that out of the box of this french site, we should not bother anymore : too complicated for small use case, i think.

**Comment by wolfgang-ch** [comment]
This map shows all "tours" from the GPX file

![image](https://user-images.githubusercontent.com/1283445/202125328-8c3e3fda-c39b-4713-a19a-64b17d4189d0.png)

**Reply by mff73** [reply]
Exactely 👍
